modules/k6/k6.go

```diff
@@ -17,17 +22,73 @@ type K6Container struct {
 	testcontainers.Container
 }
 
+type RemoteTestFileDescription struct {
+	Uri         url.URL
+	DownloadDir string
+	User        string
+	Password    string
+}
+
+func (d *RemoteTestFileDescription) getDownloadPath() string {
+	baseName := path.Base(d.Uri.Path)
+	return path.Join(d.DownloadDir, baseName)
+
+}
+
+func downloadFileFromDescription(d RemoteTestFileDescription) error {
+
+	client := http.Client{Timeout: time.Second * 60}
+	req, err := http.NewRequest(http.MethodGet, d.Uri.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "text/javascript")
+	// Set up HTTPS request with basic authorization.
+	if d.User != "" && d.Password != "" {
+		req.SetBasicAuth(d.User, d.Password)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	downloadedFile, err := os.Create(d.getDownloadPath())
+	if err != nil {
+		return err
+	}
+	defer downloadedFile.Close()
+
+	_, err = io.Copy(downloadedFile, resp.Body)
+	return err
+
+}
+
 // WithTestScript mounts the given script into the ./test directory in the container
 // and passes it to k6 as the test to run.
 // The path to the script must be an absolute path
 func WithTestScript(scriptPath string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
-		script := filepath.Base(scriptPath)
-		target := "/home/k6x/" + script
+
+	scriptBaseName := filepath.Base(scriptPath)
+	f, err := os.Open(scriptPath)
+	if err != nil {
+		panic("Cannot create reader for test file ")
+	}
+	return WithTestScriptReader(f, scriptBaseName)
+
+}
+
+// WithTestScriptReader copies files into the Container using the Reader API
+// The script base name is not a path, neither absolute or relative and should
+// be just the file name of the script
+func WithTestScriptReader(reader io.Reader, scriptBaseName string) testcontainers.CustomizeRequestOption {
```

Comment:
We must include these options in the docs (root-dir/docs/modules/k6.md). Let's include that it will only accept Javascript files and the configuration of the http client (timeout=60s, context type, etc) https://github.com/testcontainers/testcontainers-go/pull/2350/files#diff-19a4c472d6cc0154e74ab208f0a048d900eb0e6006b4365b853d56218d73cd5fR40-R50

The users of the module would like to know this option exists and how to use it, from our docs.